src/components/Project/Secrets/index.js

```diff
@@ -153,6 +161,7 @@ export default class Secrets extends React.Component {
       drawerOpen: false,
       dialogOpen: false,
       dirtyFormDialogOpen: false,
+      cursorStack: [],
```

Comment:
This is an alright way of handling the cursor history, however we will definitely want to expand this to allow arbitrary page navigation.

Another solution to consider is passing a "nextPage" or "previousPage" param to the server along with the currentPage cursor.